src/ray/rpc/raylet/raylet_client.h

```diff
@@ -1,88 +1,62 @@
 #include <unistd.h>
+#include <future>
 #include <mutex>
+#include <thread>
 #include <unordered_map>
 #include <vector>
 
+#include <grpcpp/grpcpp.h>
+
 #include "ray/common/status.h"
 #include "ray/common/task/task_spec.h"
+#include "src/ray/common/status.h"
+#include "src/ray/protobuf/raylet.grpc.pb.h"
+#include "src/ray/protobuf/raylet.pb.h"
+#include "src/ray/rpc/client_call.h"
 
 using ray::ActorCheckpointID;
 using ray::ActorID;
-using ray::ClientID;
 using ray::JobID;
 using ray::ObjectID;
 using ray::TaskID;
-using ray::UniqueID;
+using ray::WorkerID;
 
 using ray::Language;
 using ray::rpc::ProfileTableData;
-
-using MessageType = ray::protocol::MessageType;
-using ResourceMappingType =
-    std::unordered_map<std::string, std::vector<std::pair<int64_t, double>>>;
 using WaitResultPair = std::pair<std::vector<ObjectID>, std::vector<ObjectID>>;
 
-class RayletConnection {
- public:
-  /// Connect to the raylet.
-  ///
-  /// \param raylet_socket The name of the socket to use to connect to the raylet.
-  /// \param worker_id A unique ID to represent the worker.
-  /// \param is_worker Whether this client is a worker. If it is a worker, an
-  ///        additional message will be sent to register as one.
-  /// \param job_id The ID of the driver. This is non-nil if the client is a
-  ///        driver.
-  /// \return The connection information.
-  RayletConnection(const std::string &raylet_socket, int num_retries, int64_t timeout);
-
-  ~RayletConnection() { close(conn_); }
-  /// Notify the raylet that this client is disconnecting gracefully. This
-  /// is used by actors to exit gracefully so that the raylet doesn't
-  /// propagate an error message to the driver.
-  ///
-  /// \return ray::Status.
-  ray::Status Disconnect();
-  ray::Status ReadMessage(MessageType type, std::unique_ptr<uint8_t[]> &message);
-  ray::Status WriteMessage(MessageType type,
-                           flatbuffers::FlatBufferBuilder *fbb = nullptr);
-  ray::Status AtomicRequestReply(MessageType request_type, MessageType reply_type,
-                                 std::unique_ptr<uint8_t[]> &reply_message,
-                                 flatbuffers::FlatBufferBuilder *fbb = nullptr);
+namespace ray {
+namespace rpc {
 
- private:
-  /// File descriptor of the Unix domain socket that connects to raylet.
-  int conn_;
-  /// A mutex to protect stateful operations of the raylet client.
-  std::mutex mutex_;
-  /// A mutex to protect write operations of the raylet client.
-  std::mutex write_mutex_;
-};
+using ResourceMappingType =
+    std::unordered_map<std::string, std::vector<std::pair<int64_t, double>>>;
 
+/// Client used for communicating with raylet.
 class RayletClient {
  public:
-  /// Connect to the raylet.
+  /// Constructor for raylet client.
+  /// TODO(jzh): At present, client call manager and reply handler service are generated
+  /// in raylet client. Should add parameters to the constructor and pass them in.
```

Comment:
```suggestion
  /// in raylet client. Instead, we should add parameters to the constructor and pass them in.
```

Full sentences please.